Authentication/AuthenticationCoordinator.swift

```diff
@@ -0,0 +1,232 @@
+            - window: The window the application will use.
+            - componentsFactory: The authentication components factory used
+                for creating all components necessary in the authentication logic.
+    */
+    public init(sessionService: SessionService, window: UIWindow,
+                componentsFactory: AuthenticationComponentsFactoryType) {
+        _window = window
+        _componentsFactory = componentsFactory
+        self.sessionService = sessionService
+
+        sessionService.events.observeNext { [unowned self] event in
+            switch event {
+            case .LogIn(_): self._window.rootViewController = self._componentsFactory.createMainViewController()
+            case .SignUp(_): self._window.rootViewController = self._componentsFactory.createMainViewController()
+            case .LogOut(_): self._window.rootViewController = UINavigationController(rootViewController: self.createLoginController())
+            default: break
+            }
+        }
+    }
+
+    /**
+        Bootstraps your project with the authentication framework,
+        starting with the authentication project if no user is already logged in the session service.
+        Otherwise, it runs your project directly from starting the Main View Controller.
+    */
+    public final func start() {
+        if let _ = self.currentUser {
+            _window.rootViewController = _componentsFactory.createMainViewController()
+        } else {
+            let mainAuthenticationController = _componentsFactory.initialScreen == .Login ? createLoginController() : createSignupController()
+            _window.rootViewController = UINavigationController(rootViewController: mainAuthenticationController)
+        }
+    }
+    
+}
+
+// MARK: - Login Functions
+public extension AuthenticationCoordinator {
+    
+    /**
+         Creates the login controller to use for starting
+         the authentication process.
+         
+         - Returns: A valid login controller to use.
+     */
+    internal func createLoginController() -> LoginController {
+        let configuration = createLoginControllerConfiguration()
+        return LoginController(configuration: configuration)
+    }
+
+    /**
+         Creates the login view controller configuration
+         that the login controller will use to access the
+         login view model to use, the login view to display
+         and the delegates for events and transitions to
+         other screens.
+         
+         It uses the components returned by the
+         components factory.
+         
+         - Returns: A valid login controller configuration to use.
+     */
+    internal func createLoginControllerConfiguration() -> LoginControllerConfiguration {
+        let createLoginView: () -> LoginViewType = { [unowned self] in
+            return self._componentsFactory
+                .createLoginView(self._componentsFactory
```

Comment:
I would use intermediate instance variable instead of chainning this method invocations to improve readability.